tests/suites/test_suite_ctr_drbg.function

```diff
@@ -312,6 +325,57 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_THREADING_PTHREAD */
+void ctr_drbg_threads(data_t *expected_result, int reseed_interval)
+{
+#define THREAD_CNT 5
+    pthread_t threads[THREAD_CNT];
+
+    unsigned char out[16];
+    memset(out, 0, sizeof(out));
+
+    unsigned char entropy[1024];
+    memset(entropy, 0, sizeof(entropy));
+
+    test_offset_idx = 0;
+    test_max_idx = sizeof(entropy);
+
+    mbedtls_ctr_drbg_context ctx;
+    mbedtls_ctr_drbg_init(&ctx);
+
+    mbedtls_ctr_drbg_set_reseed_interval(&ctx, reseed_interval);
+
+    /* There are too many calls in this test to conveniently provide enough
```

Comment:
I don't understand this comment. Why is it inconvenient to provide enough (fake) entropy?

`mbedtls_test_entropy_func` is not thread-safe (it mutates a global variable). Is the reason that we don't want to bother making it thread-safe? But isn't the entropy function called with the lock held, so that it doesn't need to be thread safe?